cpp/tests/utilities_tests/device_atomics_test.cu

```diff
@@ -40,7 +41,7 @@ void gpu_atomic_test(T *result, T *data, size_t size)
         atomicAdd(&result[0], data[id]);
         atomicMin(&result[1], data[id]);
         atomicMax(&result[2], data[id]);
-        atomicAdd(&result[3], data[id]);
+        cudf::genericAtomicOperation(&result[3], data[id], cudf::DeviceSum{});
```

Comment:
Why this change? Also maybe explain in a comment why these four operations are applied to four locations of result?

Reply:
> Why this change?

I realized that there is no explicit cudf::genericAtomicOperation test case.
I've added it instead of `atomicAdd` which is already tested.

> Also maybe explain in a comment why these four operations are applied to four locations of result?

The reason for four locations of result is that the implementation for `int8_t` is different for each location and needs to be tested that Updating on different byte in same 32bit are occurred at same time.
I will add comment.